include/NovelRT/Ecs/Catalogue.h

```diff
@@ -0,0 +1,42 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_ECS_CATALOGUE_H
+#define NOVELRT_ECS_CATALOGUE_H
+
+#include "ComponentCache.h"
+#include "ComponentView.h"
+#include "EcsUtils.h"
+#include "EntityCache.h"
+
+namespace NovelRT::Ecs
+{
+    class Catalogue
+    {
+        private:
+        size_t _poolId;
+        ComponentCache& _componentCache;
+        EntityCache& _entityCache;
+        std::vector<EntityId> _createdEntitiesThisFrame;
+
+        public:
+        Catalogue(size_t poolId, ComponentCache& componentCache, EntityCache& entityCache) noexcept;
+
+        template<typename TComponent>
+        [[nodiscard]] auto GetComponentView()
+        {
+            return ComponentView<TComponent>(_poolId, _componentCache.GetComponentBuffer<TComponent>());
+        }
```

Comment:
I'm sorry if I keep forgetting - are we switching to Allman-style bracing for new files?

Reply:
Yes. :)

I did my best to write this feature under the new format, however until it is in I am bound to mess stuff up lol - I can't entirely remember every single convention so I'm hoping the new formatting stuff will just remember for me in the future